File: arrayRotation.py
# Print array after it is right rotated K times - python implementation

n = input("Enter number of array elements: ")
while not n.isnumeric():
	n = input("That is not a number, try again: ")

k = input("Enter number of rotations: ")
while not k.isnumeric():
	k = input("That is not a number, try again: ")

n = int(n)
k = int(k)

myArray = []

for i in range(n):
	myArray.append(i)

print("Original array: " + str(myArray))

# rotate k mod n times
for i in range(k % n): 
	# rearrange array elements
	first = myArray[0]
	for j in range(len(myArray)-1):
		myArray[j] = myArray[j+1]
	myArray[-1] = first
	print("Rotation " + str(i+1) + ": " + str(myArray))

print("Rotated array: " + str(myArray))

File: arrayRotation/arrayRotation.c
// Print array after it is right rotated K times - c implementation

#include <stdio.h>
#include <stdlib.h>

int main() {
    int k, rotations;
    int i, j;
    int arraySize;
    int* myArray;
    int* rotatedArray;

    printf("Enter array size: ");
    scanf("%d", &arraySize);

    printf("Enter number of rotations: ");
    scanf("%d", &k);

    // populate array
    myArray = (int*)malloc(arraySize * sizeof(int));
    for (i = 0; i < arraySize; i++) {
        myArray[i] = i+1;
    }

    // rotate
    rotations = k % arraySize;
    rotatedArray = (int*)malloc(arraySize * sizeof(int));
    j = 0;

    for (i = rotations; i < arraySize; i++) {
        rotatedArray[j] = myArray[i];
        j++;
    }

    for (i = 0; i < rotations; i++) {
        rotatedArray[j] = myArray[i];
        j++;
    }

    // print arrays
    printf("Original array: ");
    for (i = 0; i < arraySize; i++) {
        printf("%d ", myArray[i]);
    }

    printf("\nArray after %d rotations: ", k);
    for (i = 0; i < arraySize; i++) {
        printf("%d ", rotatedArray[i]);
    }

    free(myArray);
    free(rotatedArray);
    
    return 0;
}
